components/BottomAppBar/src/private/MDCBottomAppBarLayer.m

```diff
@@ -36,228 +36,81 @@ + (instancetype)layer {
-    kEndAngle = (CGFloat)asin(kMDCBottomAppBarFloatingButtonPositionY /
-                              kMDCBottomAppBarFloatingButtonRadius);
-    kStartAngle = (CGFloat)(M_PI - kEndAngle);
-  });
+  CGFloat arcRadius =
+      CGRectGetHeight(floatingButton.bounds) / 2 + kMDCBottomAppBarFloatingButtonRadiusOffset;
+  CGFloat navigationBarYOffset = CGRectGetMinY(navigationBarFrame);
+  CGFloat halfAngle = acos((navigationBarYOffset - floatingButton.center.y) / arcRadius);
+  CGFloat startAngle = M_PI / 2 + halfAngle;
+  CGFloat endAngle = M_PI / 2 - halfAngle;
+  CGFloat halfOfHypotenuseLength = sin(halfAngle) * arcRadius;
 
   CGFloat width = CGRectGetWidth(rect);
   CGFloat height = CGRectGetHeight(rect);
-  CGFloat midX = CGRectGetMidX(rect);
 
-  // Paths generated below differ based on the placement of the floating button.
-  switch (floatingButtonPosition) {
-    case MDCBottomAppBarFloatingButtonPositionLeading: {
-      if (layoutDirection == UIUserInterfaceLayoutDirectionRightToLeft) {
-        [self pathWithCutRight:bottomBarPath
-                         width:width
-                        height:height
-                    startAngle:kStartAngle
-                      endAngle:kEndAngle];
-      } else {
-        [self pathWithCutLeft:bottomBarPath
-                        width:width
-                       height:height
-                   startAngle:kStartAngle
-                     endAngle:kEndAngle];
-      }
-      break;
-    }
-    case MDCBottomAppBarFloatingButtonPositionCenter: {
-      [bottomBarPath moveToPoint:CGPointMake(0, kMDCBottomAppBarYOffset)];
-      CGPoint centerPath = CGPointMake(midX,
-                                       kMDCBottomAppBarYOffset -
-                                       kMDCBottomAppBarFloatingButtonPositionY);
-      [bottomBarPath addArcWithCenter:centerPath
-                               radius:kMDCBottomAppBarFloatingButtonRadius
-                           startAngle:kStartAngle
-                             endAngle:kEndAngle
-                            clockwise:NO];
-      [bottomBarPath addLineToPoint:CGPointMake(width, kMDCBottomAppBarYOffset)];
-      [bottomBarPath addLineToPoint:CGPointMake(width, height * 2)];
-      [bottomBarPath addLineToPoint:CGPointMake(0, height * 2)];
-      [bottomBarPath closePath];
-      break;
-    }
-    case MDCBottomAppBarFloatingButtonPositionTrailing: {
-      if (layoutDirection == UIUserInterfaceLayoutDirectionRightToLeft) {
-        [self pathWithCutLeft:bottomBarPath
-                        width:width
-                       height:height
-                   startAngle:kStartAngle
-                     endAngle:kEndAngle];
-      } else {
-        [self pathWithCutRight:bottomBarPath
-                         width:width
-                        height:height
-                    startAngle:kStartAngle
-                      endAngle:kEndAngle];
```

Comment:
Awesome - it looks like you made the path logic more generic. Please include a summary of this work in the PR's description.

Reply:
Summary added in the description, and added the screenshots. Sorry for missing those because I didn't think about the split PR should follow the rules strictly. Thanks for reminding it!